.gitlab-ci.yml

```diff
@@ -42,6 +42,26 @@ lint:
   script:
     - make verify
 
+nogpu-tests:
+  stage: test
+  image: $DOCKER_TAG
+  needs: ["lint"]
+  tags:
+    - timemachine
+  rules:
+    - if: $CI_EXTERNAL_PULL_REQUEST_IID
+  script:
+    - CMAKE_BUILD_PARALLEL_LEVEL=$(nproc) pip install -e .[test]
+    # HACK: remove C++/CUDA library (is there a way to tell pip not to build it?)
+    - rm timemachine/lib/custom_ops.*.so
```

Comment:
This is unfortunate but works. I couldn't find a way to install the package and test dependencies but skip building the extension with pip

Reply:
Can you just modify setup.py to skip the ext_module if an env var is set?

```
kwargs = {}
if not os.environ.get('SKIP_CUSTOM_OPS'):
    kwargs['ext_modules'] = [CMakeExtension("timemachine.lib.custom_ops", "timemachine/cpp")]
```

Reply:
> but could maybe use the optional argument in Extension?

Nice, this seems better than what I had. It doesn't seem quite right to make the extension optional though, since much of the API will not function without the extension and we haven't really implemented proper error messages for this case.

> Can you just modify setup.py to skip the ext_module if an env var is set?

Thanks, I like this approach! Makes it a deliberate action to disable building the extension, so it seems less potentially confusing to users than making the extension optional. Changed in afd2eed.